pkg/webhook/virtualmachine/virtualmachine.go

```diff
@@ -110,7 +110,21 @@ func (a *virtualMachineAnnotator) mutateCreateVirtualMachinesFn(ctx context.Cont
 		"virtualMachineName", virtualMachine.Name,
 		"virtualMachineNamespace", virtualMachine.Namespace)
 
-	return a.poolManager.AllocateVirtualMachineMac(virtualMachine)
+	previousVirtualMachine := &kubevirt.VirtualMachine{}
+	err := a.client.Get(context.TODO(), client.ObjectKey{Namespace: virtualMachine.Namespace, Name: virtualMachine.Name}, previousVirtualMachine)
+	if err != nil {
+		// Check if the object already exist.
```

Comment:
It does the opposite. Maybe describe what are you trying to do there. "If the VM does not exist yet, allocate a new MAC address"